### PR TITLE
Implement send-only interfaces

### PR DESCRIPTION
**Pull request by KatoStoelen**
This is an attempt to solve #23.

Will definately need some feedback/help to make sure that this works as I intended :) and that code standards/design principles are followed.

Initial testing looks good.

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@KatoStoelen Thanks for the PR ❤️ I hope to get to review it today or tomorrow the latest.

File: src/NServiceBus.Router/SendOnlyInterface.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+using NServiceBus.Raw;
+using NServiceBus.Transport;
+
+class SendOnlyInterface<T> : Interface where T : TransportDefinition, new()
+{
+    public SendOnlyInterface(string endpointName, string interfaceName, Action<TransportExtensions<T>> transportCustomization, Func<IRawEndpoint, IRuleCreationContext> ruleCreationContextFactory)
+    {
+        this.ruleCreationContextFactory = ruleCreationContextFactory;
+        Name = interfaceName;
+        rawConfig = new SendOnlyRawEndpointConfig<T>(endpointName, ext =>
```
Comment (SzymonPobiega) [inline src/NServiceBus.Router/SendOnlyInterface.cs]: Can `RawEndpointConfiguration.CreateSendOnly` be used here directly?

File: src/NServiceBus.Router/InterfaceConfiguration.cs [hunk 2]
```diff
@@ -6,11 +6,22 @@
     using Transport;
     using Unicast.Subscriptions.MessageDrivenSubscriptions;
 
+    /// <summary>
+    /// Represents an interface configuration,
+    /// </summary>
+    public interface IInterfaceConfiguration
```
Comment (SzymonPobiega) [inline src/NServiceBus.Router/InterfaceConfiguration.cs]: Why do we need this one?

**Comment by SzymonPobiega** [comment]
Hey. I looked at the code and it looks good ❤️ . I was just wondering if we could remove these two things?

File: src/NServiceBus.Router/SendOnlyInterface.cs (hunk 1 above)
Comment (KatoStoelen) [reply src/NServiceBus.Router/SendOnlyInterface.cs]: Probably! I'll look into it.

File: src/NServiceBus.Router/InterfaceConfiguration.cs (hunk 2 above)
Comment (KatoStoelen) [reply src/NServiceBus.Router/InterfaceConfiguration.cs]: It was mainly to reuse the extension methods in [MessageDrivenSubscriptionExtensions.cs](https://github.com/SzymonPobiega/NServiceBus.Router/blob/master/src/NServiceBus.Router/MessageDrivenSubscriptionExtensions.cs).

Didn't want to refactor that much, though I'm sure there are opportunities here and there (inheritance etc.).

I can remove `IInterfaceConfiguration` if you'd like, and add an additional set of extension methods for `SendOnlyInterface<T>`?

File: src/NServiceBus.Router/SendOnlyInterface.cs (hunk 1 above)
Comment (KatoStoelen) [reply src/NServiceBus.Router/SendOnlyInterface.cs]: https://github.com/SzymonPobiega/NServiceBus.Router/pull/24/commits/ed306de90a9b9074a9123164d617c077a1b1d3eb

File: src/NServiceBus.Router/InterfaceConfiguration.cs (hunk 2 above)
Comment (KatoStoelen) [reply src/NServiceBus.Router/InterfaceConfiguration.cs]: https://github.com/SzymonPobiega/NServiceBus.Router/pull/24/commits/7fa912b65f38bba730234a60edc8f56eb2ce576d

**Comment by SzymonPobiega** [comment]
Thanks @KatoStoelen ❤️ I just released it as https://github.com/SzymonPobiega/NServiceBus.Router/releases/tag/3.7.0. I also added the `SendOnlyInterface` concept to make it more explicit that we are dealing with different concepts.

**Pull request merged**